planning/behavior_path_planner/src/scene_module/pull_over/pull_over_module.cpp

```diff
@@ -96,115 +188,424 @@ bool PullOverModule::isExecutionRequested() const
+          parameters_.pull_over_minimum_velocity));
+        for (auto & p : status_.path.points) {
+          p.point.longitudinal_velocity_mps = std::min(p.point.longitudinal_velocity_mps, vel);
+        }
+      }
+      status_.has_decided_velocity = true;
+    }
+
+    if (status_.path_type == PathType::ARC_FORWARD || status_.path_type == PathType::ARC_BACKWARD) {
+      if (
+        isActivated() && hasFinishedCurrentPath() &&
+        (clock_->now() - *last_approved_time_).seconds() >
+          planner_data_->parameters.turn_light_on_threshold_time) {
+        parallel_parking_planner_.incrementPathIndex();
+        status_.path = parallel_parking_planner_.getCurrentPath();
+      }
+    }
+  } else {  // Replan shift -> arc forward -> arc backward path with each goal candidate.
+    // Research goal when enabling research and final path has not been decieded
+    if (parameters_.enable_goal_research) researchGoal();
+
+    status_.path_type = PathType::NONE;
+    status_.is_safe = false;
+
+    bool has_found_safe_path = false;
+    if (parameters_.search_priority == "efficient_path") {
+      has_found_safe_path = planWithEfficientPath();
+    } else if (parameters_.search_priority == "close_goal") {
+      has_found_safe_path = planWithCloseGoal();
+    } else {
+      RCLCPP_ERROR(
+        getLogger(), "search_priority shoudbe efficient_path or close_goal, but %s is given.",
+        parameters_.search_priority.c_str());
+    }
+
+    // Decelerate before the minimum shift distance from the goal search area.
+    if (has_found_safe_path) {
+      const Pose goal_pose = getRefinedGoal();
+      const auto arc_coordinates =
+        lanelet::utils::getArcCoordinates(status_.current_lanes, goal_pose);
+      const Pose search_start_pose = calcOffsetPose(
+        goal_pose, -parameters_.backward_goal_search_length, -arc_coordinates.distance, 0);
+      status_.path = util::setDecelerationVelocity(
+        status_.path, parameters_.pull_over_velocity, search_start_pose,
+        -calcMinimumShiftPathDistance(), parameters_.deceleration_interval);
+    }
+  }
+
+  BehaviorModuleOutput output;
+  output.path = status_.is_safe ? std::make_shared<PathWithLaneId>(status_.path)
+                                : std::make_shared<PathWithLaneId>(getStopPath());
+
+  publishDebugData();
+
+  // set hazard and turn singnal
+  if (status_.has_decided_path) {
+    const auto hazard_info = getHazardInfo();
+    const auto turn_info = getTurnInfo();
+
+    if (hazard_info.first.command == HazardLightsCommand::ENABLE) {
+      output.turn_signal_info.hazard_signal.command = hazard_info.first.command;
+      output.turn_signal_info.signal_distance = hazard_info.second;
+    } else {
+      output.turn_signal_info.turn_signal.command = turn_info.first.command;
+      output.turn_signal_info.signal_distance = turn_info.second;
+    }
+  }
+
+  // For experimets
+  if (parameters_.print_debug_info) {
+    printParkingPositionError();
   }
 
```

Comment:
```
updateRTCStatus(distance);
```

Reply:
fix in https://github.com/autowarefoundation/autoware.universe/pull/873/commits/39d02a128edc284dd7ef1052a78f03fe257e65bf